src/views/bootablevolumes/actions/components/EditBootableVolumesModal.tsx

```diff
@@ -41,51 +41,106 @@ const EditBootableVolumesModal: FC<EditBootableVolumesModalProps> = ({
     () => Object.keys(convertResourceArrayToMap(preferences)).sort((a, b) => a.localeCompare(b)),
     [preferences],
   );
-  const initialMetadata: BootableVolumeMetadata = useMemo(() => {
+
+  const instanceTypesAndSizesObject = useMemo(
+    () => getInstanceTypesWithSizes(instanceTypesNames),
+    [instanceTypesNames],
+  );
+
+  const initialParams = useMemo(() => {
+    const instanceTypeLabel =
+      dataSource?.metadata?.labels?.[DEFAULT_INSTANCETYPE_LABEL]?.split('.');
+
     return {
-      labels: dataSource?.metadata?.labels,
-      annotations: dataSource?.metadata?.annotations,
+      preference: dataSource?.metadata?.labels?.[DEFAULT_PREFERENCE_LABEL],
+      instanceType: instanceTypeLabel?.[0],
+      size: instanceTypeLabel?.[1],
+      description: dataSource?.metadata?.annotations?.[ANNOTATIONS.description],
     };
   }, [dataSource]);
-  const [metadata, setMetadata] = useState<BootableVolumeMetadata>(initialMetadata);
 
-  const setBootableVolumeMetadata = (key: string, fieldKey: string) => (value: string) =>
-    setMetadata((prevState) => ({
-      ...prevState,
-      [key]: { ...prevState[key], [fieldKey]: value },
-    }));
+  const [preference, setPreference] = useState<string>(initialParams.preference);
+  const [instanceType, setInstanceType] = useState<string>(initialParams.instanceType);
+  const [size, setSize] = useState<string>(initialParams.size);
+  const [sizeOptions, setSizeOptions] = useState<string[]>(
+    instanceTypesAndSizesObject[instanceType] || [],
+  );
+  const [description, setDescription] = useState<string>(initialParams.description);
+
+  const onInstanceTypeChange = (value: string) => {
```

Comment:
I recommend using a name more specific than `value` to indicate what information it carries. I assume these are `InstanceTypeCategory` values? If so, you can also try using that for the type to provide more information.